### ISSUE TITLE
keras gave bad result with too low Accuracy(~40%) using example data (iris.csv)

### ISSUE THREAD

**Issue by Zefeng-Wu** [fchollet/keras#7196]
##my code is as followings, with  keras version is 2.05

import numpy
import pandas
from keras.models import Sequential
from keras.layers import Dense
from keras.wrappers.scikit_learn import KerasClassifier
from keras.utils import np_utils
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from sklearn.preprocessing import LabelEncoder
from sklearn.pipeline import Pipeline
# fix random seed for reproducibility
seed =7
numpy.random.seed(seed)
# load dataset
dataframe = pandas.read_csv("iris.csv", header=None)
dataset = dataframe.values
X = dataset[:,0:4].astype(float)
Y = dataset[:,4]
# encode class values as integers
encoder = LabelEncoder()
encoder.fit(Y)
encoded_Y = encoder.transform(Y)
# convert integers to dummy variables (i.e. one hot encoded)
dummy_y = np_utils.to_categorical(encoded_Y)
# define baseline model
def baseline_model():
# create model
    model = Sequential()
    model.add(Dense(4, input_dim=4 , activation= "relu" ))
    model.add(Dense(3, activation= "sigmoid" ))
    # Compile model
    model.compile(loss= "categorical_crossentropy" , optimizer= "adam" , metrics=["accuracy"])
    return model
estimator = KerasClassifier(build_fn=baseline_model, nb_epoch=150, batch_size=5, verbose=0)
kfold = KFold(n_splits=10, shuffle=True, random_state=seed)
results = cross_val_score(estimator, X, dummy_y, cv=kfold)
print("Accuracy: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))

Accuracy: 47.33% (19.88%)

**Comment by bermeitinger-b** [comment]
Which accuracy are you expecting? Do you have a reference where your model actually performs better?
You can achieve above 90% accuracy by adding two hidden layers with 100 nodes each.

```python
def deeper_model():
    model = Sequential()
    model.add(Dense(4, input_dim=4 , activation= "relu" ))
    model.add(Dense(100, activation="relu"))
    model.add(Dense(100, activation="relu"))
    model.add(Dense(3, activation="sigmoid"))
    model.compile(loss="categorical_crossentropy" , optimizer="adam" , metrics=["accuracy"])
    return model
```
This will print `Accuracy: 90.00% (11.64%)`.

**Comment by u0509421** [comment]
estimator = KerasClassifier(build_fn=baseline_model, epochs=200, batch_size=5)
kfold = KFold(n=len(X), n_folds=10, shuffle=True, random_state=seed)

You can change your estimator and kfold and get 90%+ accuracy as expected.
